components/service/location/src/main/java/mozilla/components/service/location/MozillaLocationService.kt

```diff
@@ -0,0 +1,162 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.location
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.NONE
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import mozilla.components.Build
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Headers
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import mozilla.components.concept.fetch.isSuccess
+import mozilla.components.support.base.log.logger.Logger
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+
+private const val GEOIP_SERVICE_URL = "https://location.services.mozilla.com/v1/"
+private const val CONNECT_TIMEOUT_SECONDS = 10L
+private const val READ_TIMEOUT_SECONDS = 10L
+private const val USER_AGENT = "MozAC/" + Build.version
+private const val EMPTY_REQUEST_BODY = "{}"
+private const val CACHE_FILE = "mozac.service.location.region"
+private const val KEY_COUNTRY_CODE = "country_code"
+private const val KEY_COUNTRY_NAME = "country_name"
+
+/**
+ * The Mozilla Location Service (MLS) is an open service which lets devices determine their location
+ * based on network infrastructure like Bluetooth beacons, cell towers and WiFi access points.
+ *
+ * - https://location.services.mozilla.com/
+ * - https://mozilla.github.io/ichnaea/api/index.html
+ *
+ * Note: Accessing the Mozilla Location Service requires an API token:
+ * https://location.services.mozilla.com/contact
+ *
+ * @param client The HTTP client that this [MozillaLocationService] should use for requests.
+ * @param apiKey The API key that is used to access the Mozilla location service.
+ * @param serviceUrl An optional URL override usually only needed for testing.
+ */
+class MozillaLocationService(
+    private val context: Context,
+    private val client: Client,
```

Comment:
Should we make this a Lazy<Client>, similar to glean/megazord to delay initializing until needed?

Reply:
Since we last looked at this I am not sure if this will actually help?

If I do:

```Kotlin
val service = MozillaLocationService(context, lazy { components.core.client })
```

then doesn't that mean `client` is already created in order to wrap it? 🤔